mempool/src/lib.rs

```diff
@@ -126,8 +139,16 @@ impl<'env> Mempool<'env> {
             if recipient_account.account_type() != transaction.recipient_type {
                 return ReturnCode::Invalid;
             }
-            if let Err(_) = recipient_account.with_incoming_transaction(&transaction, block_height) {
-                return ReturnCode::Invalid;
+
+            match recipient_account.with_incoming_transaction(&transaction, block_height) {
+                Err(_) => return ReturnCode::Invalid,
+                Ok(r) => {
+                    // Check recipient account against filter rules
```

Comment:
Most of our comments end with a `.`.
```suggestion
                    // Check recipient account against filter rules.
```